data/campaigns/tutorial/lua/character_selection.lua

```diff
@@ -20,6 +20,7 @@ function wml_actions.select_character()
 	if character == 2 then
 		wesnoth.units.to_map({
 			type = "Fighteress",
+			side = 1,
```

Comment:
The commit says that this fixes a warning, but I couldn't reproduce a warning in the engine. Is it a warning shown only in an IDE using the EmmyLua extension?

I feel this one's worth cherry-picking, because it's good for the tutorial to be example code that UMC authors can use as a reference, so it's good for it to show how to put the character on other sides too. I'll happily do that cherry-pick, if you tell me enough to clarify the commit message's comment about warnings.

Reply:
Okay, I'll explain what happened.

The `to_map` function was documented as taking only a `unit` as a parameter, which meant that any time someone passed a table, the Lua extension decided that _every field of a unit was a mandatory key_. That seemed wrong, so I made most of the keys optional in commit 3bf903e, but left the core fields of `type` and `side` mandatory.

However, I later realized that that produces _different_ warnings, because it now considers all those fields to be possibly `nil` when _reading_ from a unit. So I'm planning to revert that change and instead annotate that `to_map` can take either a `unit` or a `WMLTable`. This means it just won't check the contents of the table, which is maybe not the best outcome, but… we'd basically have to define a separate `unitWML` type to make it actually check correctly, and even then it wouldn't be perfect, as it can't validate WML children. I _might_ do that as well, not sure.

Reply:
Cherry-picked as #8405